key.go

```diff
@@ -0,0 +1,39 @@
+package dataloader
+
+// Key is the interface that all keys need to implement
+type Key interface {
+	// Key returns a guaranteed unique string that can be used to identify an object
+	Key() string
+	// Raw returns the raw, underlaying value of the key
+	Raw() interface{}
+}
+
+// KeyList is a slice of keys with some convience methods
```

Comment:
I like `type Keys` much better than `type KeyList`! 

We should update the comment string to match the type name.

```diff
- // KeyList is a slice of keys with some convience methods.
+ // Keys wraps a slice of Key types to provide some convenience methods.